k8s/client.go

```diff
@@ -871,3 +880,35 @@ func (c *Client) GetHelmState(ctx context.Context, namespace string, secretName
 		Values:  values,
 	}, nil
 }
+
+// CreateEphemeralContainer will create a EphemeralContainer (debug container) in the specified pod.
+// EphemeralContainers are special containers which can be added after-the-fact in running pods. They're
+// useful for debugging, either when the target container image doesn't have necessary tools, or because
```

Comment:
> when the target container image doesn't have necessary tools

This could be a good usage for us going forward, a few customers are asking if cilium agent can be with scratch/distroless/minimal base image. Right now, we rely on installing new tools (e.g. tcpdump) some one cases during troubleshooting.

Reply:
Indeed, it can be a big space savings, but it will be a long time before we can rely on that feature existing. Right now it's in beta, auto-enabled but behind an optional feature gate. It is scheduled to go GA in 1.25.